tests/e2e/gc_block_volume_expansion.go

```diff
@@ -1334,6 +1334,309 @@ var _ = ginkgo.Describe("[csi-guest] Volume Expansion Test", func() {
+
+		}
+		framework.Logf("online volume expansion is successful")
+	})
+
+	/*
+		Online volume Resize beyond storage policy quota fails
+		Steps:
+		1. Create a SC with allowVolumeExpansion set to 'true'
+		2. set a quota limit on the storage policy in VC
+		3. create a PVC using the SC created in step 1 and wait for binding with PV
+		4. Create POD using above PVC
+		5. resize GC PVC to size bigger than what storage policy quota allows
+		6. Verify resize fails
+		7. Increase quota on the storage policy to allow the exapnsion
+		8. Verify SVC and PVC reach "FilesystemResizePending" state and PVs have the new size used in step 3
+		9. Check using CNS query that size has got updated to what was used in step 4
+		10. Verify size of PVs in SVC and GC to same as the one used in the step 4
+		11. delete PVC created in step 2
+		12. delete SC created in step 1
+	*/
+	ginkgo.It("Online resize beyond storage policy quota fails", func() {
+		var originalSizeInMb, fsSize int64
+
+		ctx, cancel := context.WithCancel(context.Background())
+		defer cancel()
+
+		ginkgo.By("Verify online expansion when resize beyond storage policy quota fails")
+		// Create a POD to use this PVC, and verify volume has been attached
+		ginkgo.By("Creating pod to attach PV to the node")
+		pod, err := createPod(client, namespace, nil, []*v1.PersistentVolumeClaim{pvclaim}, false, execCommand)
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+		defer func() {
+			// Delete POD
+			ginkgo.By(fmt.Sprintf("Deleting the pod %s in namespace %s", pod.Name, namespace))
+			err := fpod.DeletePodWithWait(client, pod)
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+			isDiskDetached, err := e2eVSphere.waitForVolumeDetachedFromNode(client, pv.Spec.CSI.VolumeHandle, pod.Spec.NodeName)
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+			gomega.Expect(isDiskDetached).To(gomega.BeTrue(), fmt.Sprintf("Volume %q is not detached from the node %q", pv.Spec.CSI.VolumeHandle, pod.Spec.NodeName))
+
+		}()
+
+		ginkgo.By(fmt.Sprintf("Verify volume: %s is attached to the node: %s", pv.Spec.CSI.VolumeHandle, pod.Spec.NodeName))
+		vmUUID, err := getVMUUIDFromNodeName(pod.Spec.NodeName)
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		isDiskAttached, err := e2eVSphere.isVolumeAttachedToVM(client, volHandle, vmUUID)
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		gomega.Expect(isDiskAttached).To(gomega.BeTrue(), "Volume is not attached to the node")
+
+		//Fetch original FileSystemSize
+		ginkgo.By("Verify filesystem size for mount point /mnt/volume1 before expansion")
+		originalSizeInMb, err = getFSSizeMb(f, pod)
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+		ginkgo.By("Set quota in SVC for 5Gi on policy(SC) - " + storagePolicyName)
+		svcClient, svcNamespace := getSvcClientAndNamespace()
+		createResourceQuota(svcClient, svcNamespace, "5Gi", storagePolicyName)
+		defer deleteResourceQuota(svcClient, svcNamespace)
```

Comment:
I would suggest to add this cleanup call in `AfterEach()` as well

Reply:
In this test i am setting qiota to  minimal , Hence removing it in defer to set it back to normal.  Other tests we are not doing anything with quota.